src/main/java/com/puppycrawl/tools/checkstyle/ant/CheckstyleAntTask.java

```diff
@@ -296,9 +307,9 @@ public void execute() {
             log("compiled on " + compileTimestamp, Project.MSG_VERBOSE);
 
             // Check for no arguments
-            if (fileName == null && fileSets.isEmpty()) {
+            if (fileName == null && fileSets.isEmpty() && sourcePaths.isEmpty()) {
```

Comment:
please, make it as:

```java
if (condition1
    && condition2
    && condition3) {
}
```

it will help to the future contributor to analyze coverage reports.